### PR TITLE
[WIP] scratch refactoring of using name of job directly

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -90,6 +90,7 @@ our @EXPORT = qw(
   hashwalker
   read_test_modules
   feature_scaling
+  label
```
Comment (Martchus) [inline lib/OpenQA/Utils.pm]: This name is to generic to export it by a module which we import almost everywhere.

Comment (okurz) [reply lib/OpenQA/Utils.pm]: job_label?

File: lib/OpenQA/Utils.pm [hunk 2]
```diff
@@ -1078,4 +1079,12 @@ sub any_array_item_contained_by_hash {
 
 sub base_host { Mojo::URL->new($_[0])->host || $_[0] }
 
+sub job_label {
+    my ($job) = @_;
+
+    my $test    = $job->TEST;
+    my $machine = $job->MACHINE;
+    return $machine ? "$test\@$machine" : $test;
+}
```
Comment (Martchus) [inline lib/OpenQA/Utils.pm]: This function should be defined within the DBIx result class. I don't get why you want to move it here.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 3]
```diff
@@ -273,21 +273,6 @@ sub name {
...
 
-sub label {
-    my ($self) = @_;
-
-    my $test    = $self->TEST;
-    my $machine = $self->MACHINE;
-    return $machine ? "$test\@$machine" : $test;
-}
-
-sub scenario {
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/Jobs.pm]: As far as I remember this is used for the scenario description so there should be *some* replacement.

File: lib/OpenQA/Schema/Result/JobTemplates.pm [hunk 4]
```diff
@@ -65,6 +65,10 @@ __PACKAGE__->has_many(
     settings        => 'OpenQA::Schema::Result::JobTemplateSettings',
     job_template_id => {order_by => {-asc => 'key'}});
 
+sub name {
+    $self->get_column('name') || $self->test_suite->name;
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/JobTemplates.pm]: I think having this function makes sense but I'd prefer if it had an explicit `return`. And have you tested that overriding works? If not you maybe need to define the override as `around 'name' => sub {`.

**Pull request closed without merging**